core/web/job_specs_controller.go

```diff
@@ -62,10 +62,8 @@ func (jsc *JobSpecsController) Show(c *gin.Context) {
 		jsonAPIError(c, http.StatusNotFound, errors.New("JobSpec not found"))
 	} else if err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if runs, err := jsc.App.GetStore().JobRunsFor(j.ID); err != nil {
```

Comment:
I think using separate endpoints is the way to go. With that being said, I don't like the fact that we have an ORM call that could mistakenly cripple our performance:

core/store/orm/orm.go:368
```
// JobRunsFor fetches all JobRuns with a given Job ID,
// sorted by their created at time.
func (orm *ORM) JobRunsFor(jobSpecID string) ([]models.JobRun, error) {
	runs := []models.JobRun{}
	err := orm.preloadJobRuns().
		Where("job_spec_id = ?", jobSpecID).
		Order("created_at desc").
		Find(&runs).Error
	return runs, err
}
```

Can you convert that method to take a second parameter `limit ...int` that defaults to 100?

Reply:
Can you elaborate more on how this will affect performance? `JobRunsFor` is no longer called, so what exactly becomes the new offender?

Reply:
Is JobRunsFor used anywhere else other than tests?

Reply:
Nevermind, I like your suggestion